app/src/main/java/com/soen490chrysalis/papilio/viewModel/UserProfileViewModelFactory.kt

```diff
@@ -18,9 +20,12 @@ class UserProfileViewModelFactory : ViewModelProvider.NewInstanceFactory()
 
         // Initialize Firebase Auth and inject it into the user repository
         val firebaseAuth = FirebaseAuth.getInstance()
-
+        //performance testing for retrofit service
+        val myTrace = Firebase.performance.newTrace("user_api_retrofit_service")
```

Comment:
Also, I would use another name for the trace like _retrofit-service-user-api-boot-time_